src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -229,8 +229,54 @@ chip::Protocols::InteractionModel::Status
 MatterDoorLockClusterServerPreAttributeChangedCallback(const chip::app::ConcreteAttributePath & attributePath,
                                                        EmberAfAttributeType attributeType, uint16_t size, uint8_t * value)
 {
-    // TODO: Implement attribute changes
-    return chip::Protocols::InteractionModel::Status::Success;
+    chip::Protocols::InteractionModel::Status res;
+
+    switch (attributePath.mAttributeId)
+    {
+    case chip::app::Clusters::DoorLock::Attributes::Language::Id:
+        res = emberAfPluginDoorLockOnLanguageChange(attributePath.mEndpointId, reinterpret_cast<const char *>(value + 1));
```

Comment:
why value  + 1?

Is this guaranteed to be a null terminated  string? The + 1 makes me think maybe it is a pascal string, in which case it would not be guaranteed to be null terminated. Could you explain in a comment if this is really correct?